include/state_machine.h

```diff
@@ -30,10 +30,17 @@ class StateMachine
   using state_iterator = std::list<State>::iterator;
 
   void constructObstacleMenu(State* from);
-  void constructWalkMenu(state_iterator lastState);
-  void constructStepMenu(state_iterator lastState);
+  void constructWalkMenu(state_iterator lastState, State* obstacles);
+  void constructStepMenu(state_iterator lastState, State* obstacles);
```

Comment:
Try to be consistent in parameter naming between definitions and declarations. In the declaration the `lastState` parameter is named `list_end`.
```suggestion
  void constructWalkMenu(state_iterator list_end, State* obstacles);
  void constructStepMenu(state_iterator list_end, State* obstacles);
```

Reply:
Good one!